File: ruche-connectee/mobile-app/lib/config/api_config.dart
class ApiConfig {
  // URL de base de l'API Spring Boot
  // En développement : localhost
  // En production : URL du serveur déployé
  static const String _baseUrlDev = 'http://localhost:8080';
  
  // Utiliser l'URL de développement par défaut
  // TODO: Adapter selon l'environnement (dev/prod)
  static const String baseUrl = _baseUrlDev;
  
  // Endpoints de l'API
  static const String apiPrefix = '/api/mobile';
  
  // Endpoints des ruches
  static const String ruchesEndpoint = '$apiPrefix/ruches';
  static const String ruchersEndpoint = '$apiPrefix/ruchers';
  static const String healthEndpoint = '$apiPrefix/ruches/health';
  static const String healthAuthEndpoint = '$apiPrefix/ruches/health/auth';
  
  // Timeout pour les requêtes
  static const Duration connectTimeout = Duration(seconds: 10);
  static const Duration receiveTimeout = Duration(seconds: 15);
  static const Duration sendTimeout = Duration(seconds: 15);
  
  // Headers par défaut
  static const Map<String, String> defaultHeaders = {
    'Content-Type': 'application/json',
    'Accept': 'application/json',
  };
  
  // Méthodes utilitaires pour construire les URLs
  static String get fullRuchesUrl => '$baseUrl$ruchesEndpoint';
  static String get fullRuchersUrl => '$baseUrl$ruchersEndpoint';
  static String get fullHealthUrl => '$baseUrl$healthEndpoint';
  static String get fullHealthAuthUrl => '$baseUrl$healthAuthEndpoint';
  
  static String rucheByIdUrl(String rucheId) => '$fullRuchesUrl/$rucheId';
  static String ruchesByRucherUrl(String rucherId) => '$fullRuchesUrl/rucher/$rucherId';
  
  // Vérifier si on est en mode développement
  static bool get isDevelopment => baseUrl == _baseUrlDev;
} 

File: src/config/api-config.ts
/**
 * Configuration de l'API Spring Boot
 */

// URL de base de l'API Spring Boot
// En développement: http://localhost:8080
// En production: remplacer par l'URL du serveur déployé
export const API_BASE_URL = import.meta.env.VITE_API_URL || 'http://localhost:8080';

// Endpoints de l'API
export const API_ENDPOINTS = {
  // Endpoints publics
  RUCHES: '/api/ruches',
  RUCHERS: '/api/ruchers',
  
  // Endpoints mobiles (avec authentification)
  MOBILE_RUCHES: '/api/mobile/ruches',
  MOBILE_RUCHERS: '/api/mobile/ruchers',
  
  // Endpoints de test (développement uniquement)
  TEST: '/api/test',
  TEST_HEALTH: '/api/test/health',
  
  // Endpoints de développement (sans authentification)
  DEV_HEALTH: '/dev/health',
  DEV_CREATE_TEST_DATA: (rucheId: string) => `/dev/create-test-data/${rucheId}`,
  DEV_GET_TEST_DATA: (rucheId: string) => `/dev/test-data/${rucheId}`,
  
  // Endpoints spécifiques
  MESURES_7_JOURS: (rucheId: string) => `/api/mobile/ruches/${rucheId}/mesures-7-jours`,
  CREER_DONNEES_TEST: (rucheId: string) => `/api/test/ruche/${rucheId}/creer-donnees-test`,
} as const;

// Headers par défaut pour les requêtes
export const DEFAULT_HEADERS = {
  'Content-Type': 'application/json',
} as const;

// Configuration des timeouts
export const API_CONFIG = {
  TIMEOUT: 30000, // 30 secondes
  RETRY_ATTEMPTS: 3,
  RETRY_DELAY: 1000, // 1 seconde
} as const;

/**
 * Utilitaire pour construire les headers avec authentification
 */
export const getAuthHeaders = (apiculteurId: string, token?: string) => {
  const headers: Record<string, string> = {
    ...DEFAULT_HEADERS,
    'X-Apiculteur-ID': apiculteurId,
  };
  
  if (token) {
    headers['Authorization'] = `Bearer ${token}`;
  }
  
  return headers;
};

/**
 * Utilitaire pour construire une URL complète
 */
export const buildApiUrl = (endpoint: string) => {
  return `${API_BASE_URL}${endpoint}`;
}; 